mmv1/third_party/terraform/tests/resource_storage_transfer_agent_pool_test.go

```diff
@@ -0,0 +1,206 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccStorageTransferAgentPool_agentPoolUpdate(t *testing.T) {
+	t.Parallel()
+
+	agentPoolName := fmt.Sprintf("tf-test-agent-pool-%s", randString(t, 10))
+	agentPoolNameUpdate := fmt.Sprintf("tf-test-agent-pool-%s", randString(t, 10))
+	displayName := fmt.Sprintf("tf-test-display-name-%s", randString(t, 10))
+	displayNameUpdate := fmt.Sprintf("tf-test-display-name-%s", randString(t, 10))
+
+	vcrTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckStorageTransferAgentPoolDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccStorageTransferAgentPool_agentPoolBasic(getTestProjectFromEnv(), agentPoolName, displayName),
+			},
+			{
+				ResourceName:            "google_storage_transfer_agent_pool.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"state"},
+			},
+			{
+				Config: testAccStorageTransferAgentPool_agentPoolBasic(getTestProjectFromEnv(), agentPoolName, displayNameUpdate),
+			},
+			{
+				ResourceName:            "google_storage_transfer_agent_pool.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"state"},
+			},
+			{
+				Config: testAccStorageTransferAgentPool_updateLimitMbps(getTestProjectFromEnv(), agentPoolName, displayNameUpdate),
+			},
+			{
+				ResourceName:            "google_storage_transfer_agent_pool.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"state"},
+			},
+			{
+				Config: testAccStorageTransferAgentPool_omitDisplayName(getTestProjectFromEnv(), agentPoolName),
+			},
+			{
+				ResourceName:            "google_storage_transfer_agent_pool.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"state"},
+			},
+			{
+				Config: testAccStorageTransferAgentPool_omitBandwidthLimit(getTestProjectFromEnv(), agentPoolName, displayNameUpdate),
+			},
+			{
+				ResourceName:            "google_storage_transfer_agent_pool.foo",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"state"},
+			},
+			{
+				Config: testAccStorageTransferAgentPool_agentPoolBasic(getTestProjectFromEnv(), agentPoolNameUpdate, displayName),
```

Comment:
Question: Do we need this test step here? Looks like it's trying update the immutable field `name` and I assume it will just delete and recreate the resource.

Reply:
Right. It will create a new resource. I am not sure if this test step is needed or not. I can remove it if it is not needed.

Reply:
Then I think we can remove this step as it's just simply testing the creation + deletion of the resource which we have got them covered in previous steps I believe :)

Reply:
That makes sense. Thanks.